hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java

```diff
@@ -117,6 +117,8 @@ public H2Dialect(int version) {
 			this.sequenceInformationExtractor = SequenceInformationExtractorNoOpImpl.INSTANCE;
 			this.querySequenceString = null;
 		}
+		// H2 doesn't allow a length for BIT
+		registerColumnType( Types.BIT, "bit" );
```

Comment:
I don't think this change is a good one. H2 doesn't allow a length in `bit` because it's a synonym for `boolean`. So if you're trying to use a `bit(10)` on H2, then that should give you an *error*.

Making this change means you would get the error later: not when the schema is exported, but when you try to write to the column. I think it's better to get the error earlier than later, so I would not make this change.

Reply:
I'll remove that.

Reply:
Note, this is kinda similar to the situation with the Transact SQL dialect above.

Reply:
To be consistent with other dialects we could map `bit` with length>1 to `smallint` if we decide that makes sense.

Reply:
I'd like to remove all those re-mappings since the BitTypeDescriptor always expects true/false i.e. it even uses `getBoolean`/`setBoolean` on the JDBC driver. Passing through the length if the DB supports it is ok I guess, but otherwise I'd just omit it. WDYT?